sentry-android-core/src/main/java/io/sentry/android/core/AndroidTransactionProfiler.java

```diff
@@ -39,10 +37,10 @@ final class AndroidTransactionProfiler implements ITransactionProfiler {
   private final @NotNull ISentryExecutorService executorService;
   private final @NotNull BuildInfoProvider buildInfoProvider;
   private boolean isInitialized = false;
-  private int transactionsCounter = 0;
+  private final @NotNull AtomicBoolean isRunning = new AtomicBoolean(false);
   private final @NotNull SentryFrameMetricsCollector frameMetricsCollector;
   private @Nullable ProfilingTransactionData currentProfilingTransactionData;
```

Comment:
should this be volatile?

Reply:
i think volatile is still needed for immediate visibility between threads? so we either need to wrap it in to a lock in `close()` or make it `volatile` I think